### DISCUSSION TITLE
Upgrade from 4.14 to 4.15

### DISCUSSION THREAD

**Discussion by dkoci** [okd-project/okd#1916] [Q&A]
Hi. Is it safe to upgrade from 4.14.0-0.okd-2024-01-26-175629 version to the latest 4.15 version (4.15.0-0.okd-2024-03-10-010116)? Or is it better to upgrade version by version (4.14.0-0.okd-2024-01-26-175629 --> 4.15.0-0.okd-2024-01-27-070424 --> 4.15.0-0.okd-2024-02-10-035534 --> ...... -> 4.15.0-0.okd-2024-03-10-010116)?

Thank you.

**Comment by codespearhead** [comment]
Yes.

You can skip patch versions when upgrading minor versions, but you shouldn't skip minor versions when upgrading minor versions or major versions, as per the official [Kubernetes Documentation](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/):

> Skipping MINOR versions when upgrading is unsupported.

Which is best exemplified in the [SUSE downstream Documentation](https://en.opensuse.org/Kubic:Upgrading_kubeadm_clusters):

> You only can upgrade from one MINOR version to the next MINOR version, or between PATCH versions of the same MINOR. That is, you cannot skip MINOR versions when you upgrade. For example, you can upgrade from 1.y to 1.y+1, but not from 1.y to 1.y+2.
